codegen-serde/src/main/kotlin/software/amazon/smithy/rust/codegen/serde/SerializeImplGenerator.kt

```diff
@@ -0,0 +1,397 @@
+     */
+    private fun serializeMember(
+        shape: MemberShape,
+        memberRef: String,
+    ): Writable {
+        val target = codegenContext.model.expectShape(shape.target)
+        return writable {
+            serializerFn(target) {
+                rust("&$memberRef")
+            }.plus { rust(".serialize_ref(&self.settings)") }(this)
+        }.letIf(target.hasTrait<SensitiveTrait>()) { memberSerialization ->
+            memberSerialization.map {
+                rustTemplate(
+                    "&#{Sensitive}(#{it}).serialize_ref(&self.settings)",
+                    *SupportStructures.codegenScope,
+                    "it" to it,
+                )
+            }
+        }
+    }
+
+    private fun structSerdeImpl(shape: StructureShape): Writable {
+        return writable {
+            implSerializeConfigured(codegenContext.symbolProvider.toSymbol(shape)) {
+                rustTemplate(
+                    """
+                    use #{serde}::ser::SerializeStruct;
+                    """,
+                    *SupportStructures.codegenScope,
+                )
+                rust(
+                    "let mut s = serializer.serialize_struct(${
+                        shape.contextName(codegenContext.serviceShape).dq()
+                    }, ${shape.members().size})?;",
+                )
+                rust("let inner = &self.value;")
+                for (member in shape.members()) {
+                    val serializedName = member.memberName.dq()
+                    val fieldName = codegenContext.symbolProvider.toMemberName(member)
+                    val field = safeName("member")
+                    val fieldSerialization =
+                        writable {
+                            rustTemplate(
+                                "s.serialize_field($serializedName, #{member})?;",
+                                "member" to serializeMember(member, field),
+                            )
+                        }
+                    if (codegenContext.symbolProvider.toSymbol(member).isOptional()) {
+                        rust("if let Some($field) = &inner.$fieldName { #T }", fieldSerialization)
+                    } else {
+                        rust("let $field = &inner.$fieldName; #T", fieldSerialization)
+                    }
+                }
+                rust("s.end()")
+            }
+        }
+    }
+
+    private fun serializeUnionImpl(shape: UnionShape): Writable {
+        val unionName = shape.contextName(codegenContext.serviceShape)
+        val symbolProvider = codegenContext.symbolProvider
+        val unionSymbol = symbolProvider.toSymbol(shape)
+
+        return writable {
+            implSerializeConfigured(unionSymbol) {
+                rustTemplate(
+                    """
+                    """,
+                    *SupportStructures.codegenScope,
+                )
```

Comment:
Is this meant to write an empty line or is it a leftover thing?

Reply:
leftover thanks